### PR TITLE
Clean up extensions on primitives

### LINKED ISSUE #752: SUSHI does not export extensions correctly when setting extension on primitive using caret rule
Easier to explain by example ([FSH Online](https://fshschool.org/FSHOnline/#/share/36L5EiC)):
```
Profile: MyProfile
Parent: Patient
// This serializes correctly:
* ^extension[MyBooleanExtension].valueBoolean = true
// But this does not serialize correctly:
* ^status.extension[MyBooleanExtension].valueBoolean = true

Extension: MyBooleanExtension
* value[x] only boolean
```

The first extension gets serialized correctly:
```json
{
  "url": "http://example.org/StructureDefinition/MyBooleanExtension",
  "valueBoolean": true
}
```

The second does not:
```json
{
  "_sliceName": "MyBooleanExtension",
  "url": {
    "assignedValue": "http://example.org/StructureDefinition/MyBooleanExtension",
    "_primitive": true
  },
  "valueBoolean": {
    "assignedValue": true,
    "_primitive": true
  }
}
```

### PR DESCRIPTION
**Pull request by mint-thompson**
Fixes #752 and completes task [CIMPL-668](https://standardhealthrecord.atlassian.net/browse/CIMPL-668).

When cleaning up properties on a StructureDefinition, properties are recursively traversed to find and remove those that were added by SUSHI to help in the export process. Properties that were added by SUSHI start with an underscore character. Properties that contain extra attributes of primitives also start with an underscore character, but will have a
sibling property with the same name without the underscore. Therefore, if such a sibling property is present, the property being examined is a true member of the StructureDefinition, and should be included in the cleanup traversal.

Basically, I'm _pretty_ sure that doing this won't ever get us into trouble, but I might be wrong.

### CODE REVIEW

**Review by ngfreiter — CHANGES_REQUESTED** [review]
I think the assumption that the sibling property will exist can get us into trouble. If you check this part of the spec: https://www.hl7.org/fhir/json.html#primitive, it does say:
> Note: If the primitive has an id attribute or extension, but no value, only the property with the _ is rendered.

And for the following FHIR:
```
Profile: MyProfile
Parent: Patient
// This serializes correctly:
* ^extension[MyBooleanExtension].valueBoolean = true
// But this does not serialize correctly:
* ^status.extension[MyBooleanExtension].valueBoolean = true
* ^publisher.extension[MyBooleanExtension].valueBoolean = true

Extension: MyBooleanExtension
* value[x] only boolean
```
This PR works for `status`, but the problem still appears on `publisher`:
```
"_`publisher`": {
    "extension": [
      {
        "_sliceName": "MyBooleanExtension",
        "url": {
          "assignedValue": "http://example.org/StructureDefinition/MyBooleanExtension",
          "_primitive": true
        },
        "valueBoolean": {
          "assignedValue": true,
          "_primitive": true
        }
      }
    ]
  }
```
because there is no value for the primitive `publisher` element. I wonder if maybe we should just make the elements that SUSHI uses for tracking start more distinctively. In retrospect it was probably a bad decision to use a character that also has meaning in the FHIR spec (pretty sure this was me, so sorry for that one).

**Comment by mint-thompson** [comment]
@ngfreiter thanks for catching that scenario. So, it seems like it'll be better to have the skip function directly check for the SUSHI-added properties like `_sliceName` and `_primitive`.

**Review by julianxcarter — APPROVED** [review]

**Review by ngfreiter — APPROVED** [review]
This looks good to me!
